src/components/common/Dialogs/ControlSafeDialog/TransactionTypesSection/TransferFundsSection.tsx

```diff
@@ -1,14 +1,239 @@
+        const baseUrl = getTxServiceBaseUrl(chainName);
+        const response = await fetch(
+          `${baseUrl}/v1/safes/${safeAddress}/balances/`,
+        );
+        if (response.status === 200) {
+          const data = (await response.json()) as SafeBalanceApiData[];
+          const formattedSafeBalances: SafeBalance[] = data.map(
+            (balanceData) => ({
+              balance: balanceData.balance,
+              token:
+                balanceData.tokenAddress && balanceData.token
+                  ? {
+                      tokenAddress: balanceData.tokenAddress,
+                      name: balanceData.token.name,
+                      symbol: balanceData.token.symbol,
+                      decimals: balanceData.token.decimals,
+                      thumbnail: balanceData.token.logoUri,
+                      type: TokenType.Erc20,
+                    }
+                  : null,
+            }),
+          );
+          setSavedTokens((tokens) => ({
+            ...tokens,
+            [safeAddress as string]: formattedSafeBalances,
+          }));
+          setSafeBalances(formattedSafeBalances);
+        }
+      } catch (e) {
+        setBalanceError(MSG.balancesError);
+        /* log.error(e); */
+      } finally {
+        setIsLoadingBalances(false);
+      }
+    }
+    // setSafeBalances causes infinite loop
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [safe, safeAddress, setSavedTokens]);
+
+  useEffect(() => {
+    if (safeAddress) {
+      const savedTokenData = savedTokens[safeAddress];
+      if (savedTokenData) {
+        setSafeBalances(savedTokenData);
+        /* setBalanceError(''); */
+      } else {
+        getSafeBalance();
+      }
+    }
+    // setSafeBalances causes infinite loop
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [safeAddress, getSafeBalance, savedTokens]);
+
+  const selectedTokenAddress =
+    transactions[transactionFormIndex].tokenData?.tokenAddress;
+
+  const selectedBalance = getSelectedSafeBalance(
+    safeBalances,
+    selectedTokenAddress,
+  );
+
+  const formattedSafeBalance = moveDecimal(
+    selectedBalance?.balance || 0,
+    -(selectedBalance?.token?.decimals || DEFAULT_TOKEN_DECIMALS),
+  );
+
+  if (isLoadingBalances) {
+    return <Loading message={MSG.balancesLoading} />;
+  }
+
+  if (balanceError) {
+    return <Error error={balanceError} />;
+  }
+
   return (
-    <DialogSection>
-      <h2>TODO</h2>
-    </DialogSection>
+    <>
+      <DialogSection>
+        <AmountBalances
+          selectedSafe={selectedSafe}
+          safeBalances={safeBalances}
+          disabledInput={disabledInput}
+          transactionFormIndex={transactionFormIndex}
+          /* handleValidation={handleValidation} */
+          /* handleChange={handleInputChange} */
+          maxButtonParams={{
+            maxAmount: `${formattedSafeBalance}`,
+            options: {},
```

Comment:
I don't think we need this `options`, right?

Reply:
on further inspection ive removed all the placeholders for manual validation checks and replaced them with these two options in the button
```ts
            options: { shouldTouch: true, shouldValidate: true },
```